client/request.go

```diff
@@ -0,0 +1,137 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/go-http-utils/headers"
+	"github.com/jtacoma/uritemplates"
+)
+
+type Request struct {
+	uriTemplate  string
+	uriVariables map[string]interface{}
+
+	method string
+	header http.Header
+	body   io.Reader
+}
+
+func NewRequest(method, uriTemplate string) *Request {
+	return &Request{
+		uriTemplate:  uriTemplate,
+		uriVariables: make(map[string]interface{}),
+
+		method: method,
+		header: make(http.Header),
+		body:   http.NoBody,
+	}
+}
+
+func Get(uriTemplate string) *Request {
+	return NewRequest(http.MethodGet, uriTemplate)
+}
+
+func Head(uriTemplate string) *Request {
+	return NewRequest(http.MethodHead, uriTemplate)
+}
+
+func Post(uriTemplate string) *Request {
+	return NewRequest(http.MethodPost, uriTemplate)
+}
+
+func Put(uriTemplate string) *Request {
+	return NewRequest(http.MethodPut, uriTemplate)
+}
+
+func Delete(uriTemplate string) *Request {
+	return NewRequest(http.MethodDelete, uriTemplate)
+}
+
+func Connect(uriTemplate string) *Request {
+	return NewRequest(http.MethodConnect, uriTemplate)
+}
+
+func Options(uriTemplate string) *Request {
+	return NewRequest(http.MethodOptions, uriTemplate)
+}
+
+func Trace(uriTemplate string) *Request {
+	return NewRequest(http.MethodTrace, uriTemplate)
+}
+
+func Patch(uriTemplate string) *Request {
+	return NewRequest(http.MethodPatch, uriTemplate)
+}
+
+func (r *Request) Assign(variable string, value interface{}) *Request {
```

Comment:
I'm thinking about this. But I don't find a better idea. Since this is very correct ^^